src/ucp/api/ucp.h

```diff
@@ -427,9 +447,33 @@ typedef struct ucp_context_attr {
      * UCP request data, which is defined by this value.
      */
     size_t                request_size;
+
+    /**
+     * This flag indicates if this context is shared by multiple workers
+     * from different threads. If so, this context needs thread safety
+     * support; otherwise, the context does not need to provide thread
+     * safety.
+     */
+    int                   mt_workers_shared;
```

Comment:
this can be just a "ucs_thread_mode_t thread_mode" field to indicate whether we eventually decided to make the context thread-safe or not. also its documentation can be only "thread safety level of the context" - same as in ucp_worker_attr